### PR TITLE
fix: skip thumbnail generation for GIF images

### LINKED ISSUE #6596: gif 生成多尺寸图片时，会仅保留第一帧
### 系统信息

- 外部访问地址: https://demo.halo.run
- 启动时间: 2024-09-04 20:00
- 版本: 2.19.0
- 构建时间: 2024-09-02 10:37
- Git Commit: 142f46b
- Java: IBM Semeru Runtime Open Edition / 21.0.4+7-LTS
- 数据库: PostgreSQL / 15.2 (Debian 15.2-1.pgdg110+1)
- 操作系统: Linux / 5.15.0-88-generic
- 已激活主题: [Nexus 知识库主题 1.2.0](https://www.halo.run/store/apps/app-WDdmT)
- 已启动插件:
  - [编辑器超链接卡片 1.0.3](https://github.com/halo-sigs/plugin-editor-hyperlink-card)

### 使用的哪种方式运行？

Docker

### 发生了什么？

比如：

Top up credit

生成其他尺寸的图片后，会仅保留第一帧（白色）。

/kind bug
/area core

### 复现步骤

_No response_

### 相关日志输出

_No response_

### 附加信息

_No response_

### PR DESCRIPTION
**Pull request by guqing**
#### What type of PR is this?
/kind bug
/area core
/milestone 2.20.x

#### What this PR does / why we need it:
修复 GIF 缩略图生成只会保留第一帧的问题

#### Which issue(s) this PR fixes:
Fixes #6596

#### Does this PR introduce a user-facing change?
```release-note
修复 GIF 缩略图生成只会保留第一帧的问题
```

### CODE REVIEW

File: application/src/main/java/run/halo/app/core/attachment/ThumbnailGenerator.java [hunk 1]
```diff
@@ -71,12 +75,14 @@ private void generateThumbnail(Path tempImagePath) throws IOException {
             throw new IOException("File size exceeds the limit: " + MAX_FILE_SIZE);
         }
         var formatNameOpt = getFormatName(file);
+        var formatName = formatNameOpt.orElse("jpg");
```
Comment (JohnNiang) [inline application/src/main/java/run/halo/app/core/attachment/ThumbnailGenerator.java:78]: 我的理解看看对不对：如果 formatName 不存在是不是也应该不支持生成缩略图呢？

Comment (guqing) [reply application/src/main/java/run/halo/app/core/attachment/ThumbnailGenerator.java]: 确实如此

**Review by JohnNiang — APPROVED** [review]
/approve

**Review by ruibaby — APPROVED** [review]
/lgtm

**Pull request merged**